Library/Homebrew/formula.rb

```diff
@@ -560,6 +573,25 @@ def self.path name
     Pathname.new("#{HOMEBREW_LIBRARY}/Formula/#{name.downcase}.rb")
   end
 
+  def self.search_path name
+    p = Formula.path(name)
+    return p if p.file?
+    name = name.downcase
+    taps = Dir["#{HOMEBREW_LIBRARY}/Taps/*/*/"]
+    p = taps.map { |tap| Dir["#{tap}#{name}.rb", "#{tap}Formula/#{name}.rb", "#{tap}HomebrewFormula/#{name}.rb"] }
+    p = p.flatten.map(&Pathname.method(:new)).select { |f| f.file? }
+    case p.size
+    when 0; nil
+    when 1; p[0]
+    else
+      odie <<-EOS.undent
```

Comment:
I'm not sure here. Should I use `odie` or `ofail` + `raise` or something else? @mikemcquaid